eclipse-theia/theia
# Implement scm.untrackedChanges preference

#### What it does

Fixes #11253 by implementing the `git.untrackedChanges` preference.


#### How to test


1. Open a workspace in a git repository.
2. Modify some files and add some files.
3. Open the SCM view
4. Test the values of the `git.untrackedChanges` preference:
 - if `mixed`, untracked and modified should be shown together, as they are on `master`.
 - if `separate`, untracked and modified should be shown separately.
 - if `hidden`, untracked files should not be shown in the SCM view. 
 5. Test the preference values in both `list` and `tree` view modes of the SCM tree.
 6. Observe that the `Stage All` command now ignores untracked files unless `git.untrackedFiles` is set to `mixed`, and is disabled if there is nothing to stage.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
